eZ/Bundle/EzPublishCoreBundle/Imagine/VariationPurger/LegacyStorageImageFileList.php

```diff
@@ -96,7 +97,7 @@ private function fetchRow(): void
         ++$this->cursor;
         $imageId = $this->rowReader->getRow();
 
-        if (substr($imageId, 0, strlen($prefix)) === $prefix) {
+        if (strpos((string)$imageId, $prefix) === 0) {
```

Comment:
As far as I know the previous variant would be _slightly_ more efficient (but that's micro). What was the reason for change?

Reply:
Symfony polyfill for PHP 7.4 proposes the following implementation of `str_starts_with`:
```php
public static function str_starts_with(string $haystack, string $needle): bool
{
    return 0 === strncmp($haystack, $needle, strlen($needle));
}
```
Benefits include:
 * binary-safe comparison
 * only beginning of the string is compared, which is not the case for `strpos` (which looks up through the whole string)